### ISSUE TITLE
Testing new server on RM2

### ISSUE THREAD

**Issue by bordaigorl** [bordaigorl/rmview#88]
The [`test-new-server`](https://github.com/bordaigorl/rmview/tree/test-new-server) branch includes the new server provided by @pl-semiotics .
This server solves the only outstanding items of #63.
In particular:
* it is compatible with updates 2.6, 2.7 and 2.8
* it is compatible with `rm2fb`

If you have a RM2, I'd be grateful if you could test the new server and report success/issues here.
Reports for RM1 are welcome too.

**Please note** the old and new servers are known not to work on the 2.9 beta.
Watch #86 for updates on that, or see #87 for using rmview with the new built-in server of 2.9.

# Instructions

1. Clone or [Download](https://github.com/bordaigorl/rmview/archive/refs/heads/test-new-server.zip) the [`test-new-server`](https://github.com/bordaigorl/rmview/tree/test-new-server) branch.

2. From the root directory of the repo, install the test version using

      pip install .

   (mind the final `.` dot)

3. if you previously installed the `/usr/lib/libcrypto.so.1.0.2` library, disable it temporarily by running

        ssh root@RM_IP mv /usr/lib/libcrypto.so.1.0.2 /usr/lib/libcrypto.so.1.0.2.bak

   If you installed a custom version of the server back that up as well, for example with

       ssh root@RM_IP 'mv $HOME/rM-vnc-server-standalone $HOME/rM-vnc-server-standalone.bak'

4. Test rmview with the new server by running

        rmview --install-server

5. Report your experience:
     - if everything works correctly, use the :+1: reaction on this post; leave a comment with RM1/2 firmware version and whether rm2fb was installed.
     - if there was some issue, react with :-1: and leave a comment below for what went wrong.

6. After you finished your tests, to go back to the previous/official version switch to the `vnc` branch and reinstall with `pip install .`. Restore the libcrypto library and old server with

        ssh root@RM_IP mv /usr/lib/libcrypto.so.1.0.2.bak /usr/lib/libcrypto.so.1.0.2
        ssh root@RM_IP 'rm $HOME/rM-vnc-server-standalone'

   At the next run of rmview you will be offered to install the server again.

**Comment by vale981** [comment]
Failed to start VNC server on reMarkable: /home/root/rM-vnc-server-standalone: error while loading shared libraries: libcrypto.so.1.0.2: cannot open shared object file: No such file or directory

RM2 with Version 2.8.0.98

works with the manually copied libcrypto

still freezes with rm2fb

**Comment by bordaigorl** [comment]
@vale981 thanks for trying. Looks like it's the old server you are testing?
Can you try removing it from the tablet with `ssh root@RM_IP 'rm $HOME/rM-vnc-server-standalone'` and auto installing it with rmview again?
You also have to make sure you are installing rmview before testing, especially if you are just switching branches and you previously installed rmview. This is because `pyrcc5` needs to be run to update the binaries in `resources.py`.

**Comment by vale981** [comment]
> You also have to make sure you are installing rmview before testing, especially if you are just switching branches and you previously installed rmview. This is because `pyrcc5` needs to be run to update the binaries in `resources.py`.

ok that may be it.
i'm on nixos though where this is not so easy. i'll try

**Comment by vale981** [comment]
i did run `setup.py build` to no avail. shouldn't that suffice?

**Comment by vale981** [comment]
ahh ok, i did run `pyrcc5` manually and now it's stuck at
![image](https://user-images.githubusercontent.com/4025991/126642971-6d35a092-5826-4570-8a29-91df3f76e07a.png)

**Comment by vale981** [comment]
Ok and now it works
Thank you SO much :)

```
RM2 with Version 2.8.0.98
with rm2fb
```

**Comment by ddund** [comment]
Tried the new server and everything seems to work fine.

RM2 v2.8.0.98 without rm2fb

**Comment by larsblumberg** [comment]
Testing this would be easier for me if I could run this in docker as I don’t want to spoil my system‘s pip.
I’ll manually apply the dockerfile introduction to this branch and report my testing results!

**Comment by bordaigorl** [comment]
@larsblumberg I see...I used the `vnc` branch as base to only test the server and not get reports of possible bugs in `devel`.
The quickest way to test without having to fiddle with rmview is to just overwrite the `$HOME/rM-vnc-server-standalone` file on your tablet with the appropriate one in the `bin` folder of the `test-new-server` branch (and don't forget to `chmod +x` it).

**Comment by vale981** [comment]
> Testing this would be easier for me if I could run this in docker as I don’t want to spoil my system‘s pip.
> I’ll manually apply the dockerfile introduction to this branch and report my testing results!

maybe virtual envs can work for you

**Comment by dsborets** [comment]
RM2 ver. 2.9.0.153 without rm2fb

```
[INFO] Connecting...
[INFO] Connected to 192.168.0.3
[INFO] Installing...
[INFO] Installation successful!
[INFO] Starting VNC server (command=$HOME/rM-vnc-server-standalone)
[INFO] Start command stdout output: peek: Input/output error
[INFO] Stopping VNC server...
[INFO] QUITTING: Fri Jul 30 21:06:01 2021
```

when I tried to run server on RM manually, I've got:

```
Using backend libqsgepaper-snoop
No cached info found for /proc/228/exe
Uncompressing extraction program
Running extraction pass 852
peek: Input/output error
open cache: No such file or directory
```

**Comment by vale981** [comment]
> RM2 ver. 2.9.0.153 without rm2fb
> 
> ```
> [INFO] Connecting...
> [INFO] Connected to 192.168.0.3
> [INFO] Installing...
> [INFO] Installation successful!
> [INFO] Starting VNC server (command=$HOME/rM-vnc-server-standalone)
> [INFO] Start command stdout output: peek: Input/output error
> [INFO] Stopping VNC server...
> [INFO] QUITTING: Fri Jul 30 21:06:01 2021
> ```
> 
> when I tried to run server on RM manually, I've got:
> 
> ```
> Using backend libqsgepaper-snoop
> No cached info found for /proc/228/exe
> Uncompressing extraction program
> Running extraction pass 852
> peek: Input/output error
> open cache: No such file or directory
> ```

2.9 has a built-in vnc server, maybe that's interfering

**Comment by bordaigorl** [comment]
@dsborets From the instructions:

> Please note the old and new servers are known not to work on the 2.9 beta.
> Watch #86 for updates on that, or see #87 for using rmview with the new built-in server of 2.9.

TLDR: the server rmview uses makes assumptions about the firmware of the tablet; 2.9 changes the firmware in ways that the server is not prepared to handle and thus it gives up. The new 2.9 also ships with a built-in server; rmview will be compatible with it once 2.9 is shipped as an official release.
